packages/components/src/custom-select-control-v2/styles.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * External dependencies
+ */
+import styled from '@emotion/styled';
+import { css } from '@emotion/react';
+// eslint-disable-next-line no-restricted-imports
+import * as Ariakit from '@ariakit/react';
+
+/**
+ * Internal dependencies
+ */
+import { COLORS } from '../utils';
+import { space } from '../utils/space';
+import type { CustomSelectProps } from './types';
+
+export const CustomSelectLabel = styled( Ariakit.SelectLabel )`
+	font-size: 11px;
+	font-weight: 500;
+	line-height: 1.4;
+	text-transform: uppercase;
+	margin-bottom: ${ space( 2 ) };
+`;
+
+const inputHeights = {
+	default: 40,
+	small: 24,
+};
+
+const getVariableInputStyles = (
+	size: NonNullable< CustomSelectProps[ 'size' ] >,
+	hasCustomRenderProp: boolean
+) => {
+	return css`
+		${ hasCustomRenderProp ? 'min-height' : 'height' }: ${ inputHeights[
+			size
+		] }px;
+		font-size: ${ size === 'small' && ! hasCustomRenderProp
+			? '11px'
+			: '13px' };
+		padding: ${ size === 'small' && ! hasCustomRenderProp
+			? space( 2 )
+			: space( 4 ) };
+	`;
+};
+
+export const CustomSelectButton = styled( Ariakit.Select )< {
```

Comment:
There is currently an error in the console about `hasCustomRenderProp` being applied to the underlying `button` component rendered by `Ariakit`.

One way to prevent this error from happening is to [tell emotion that we don't want to forward that prop](https://emotion.sh/docs/styled#customizing-prop-forwarding):

```suggestion
export const CustomSelectButton = styled( Ariakit.Select, {
	// Do not forward `hasCustomRenderProp` to the underlying Ariakit.Select component
	shouldForwardProp: ( propName ) => propName !== 'hasCustomRenderProp',
} )< {
```

Reply:
Nice catch! And thank you for the example and info on prop forwarding 😄